src/components/admin/article/ArticleController.jsx

```diff
@@ -335,35 +319,28 @@ export default class ArticleController extends FalcorController {
             floatingLabelText="Title"
             fullWidth
           />
-          <CategorySelector
+          <ListSelector
             type="Category"
-            chosenType={chosenCategory}
-            update={this.fieldUpdaters.category}
+            chosenType={this.state.category || 'none'}
+            update={this.updateCategory}
             disabled={this.state.saving}
             types={categories}
-          />
-          <TextField
-            name="image"
-            value={this.state.image}
-            floatingLabelText="Image (Remember to use https:// not http://)"
-            disabled={this.state.saving}
-            onChange={this.fieldUpdaters.image}
-            fullWidth
           /><br />
-          <TextField
+          <ImageField
+            image={this.state.image}
+            disabled={this.state.saving}
+            updateImage={this.updateImage}
+          />
+          <br />
+          <MaxLenTextField
             name="teaser"
-            floatingLabelText={
-              `Teaser (${this.state.teaser.length} of ${MAX_TEASER_LENGTH} characters)`
-            }
             value={this.state.teaser}
-            disabled={this.state.saving}
-            onChange={this.fieldUpdaters.teaser}
-            multiLine
-            rows={2}
-            fullWidth
-          /><br />
+            maxLen={156}
```

Comment:
I feel like the 156 is much more hidden now like this than when it was defined as a named constant, looks more arbitrary... What are your thoughts about this? Could we maybe put it in a constants file or... ??

Reply:
I agree this is not the solution, but I also don't like `MAX_TEASER_LENGTH` being defined like it was in `ArticleController`. I think we should either have a global const file(so that all `MAX_$FIELD_LENGTH`'s can be changed from the same place) or I can make a very small component called `TeaserField` that holds this more clearly. 

I'm partial to the former - TODO

Reply:
I'm okay with both